src/artifacts.ts

```diff
@@ -87,3 +90,79 @@ export async function fileVersion(
     throw Error(`Could not version file: ${file}`)
   }
 }
+
+export async function setDefaultIndex(inputs: InputsArtifacts): Promise<void> {
+  const client: AxiosInstance = axios.create({
+    auth: {
+      username: inputs.user,
+      password: inputs.password
+    },
+    httpsAgent: new https.Agent({
+      keepAlive: true,
+      maxSockets: 20
+    })
+  })
+  const sha: string = github.context.sha
+  const shortSha: string = sha.substring(0, 10)
+  const ref: string = github.context.ref
+    .replace('refs/heads/', '')
+    .replace('refs/tags/', '')
+  // Is the build on a tag or a branch
+  const refType: string = process.env['GITHUB_REF_TYPE'] || 'branch'
+  const metadata: object = {
+    commit: sha,
+    shortcommit: shortSha,
+    [refType]: ref
+  }
+  // Adding another set of metadata that are equal to action
+  // in terms of key naming or without any modification to the value
+  const actionsMetadata: object = {
+    ref: github.context.ref,
+    [refType]: github.context.ref,
+    sha,
+    event_name: github.context.eventName,
+    actor: github.context.actor,
+    run_number: github.context.runNumber
+  }
+  core.info('Uploading default index...')
+  await setIndex(client, inputs.url, metadata)
+  await setIndex(client, inputs.url, actionsMetadata)
+
+  core.info('Index has been uploaded')
+}
+
+export async function setIndex(
+  client: AxiosInstance,
+  url: string,
+  metadata: object
+): Promise<AxiosResponse> {
+  const owner: string = github.context.repo.owner.toLowerCase()
+  const repo: string = github.context.repo.repo.toLowerCase()
+  const runNumber: string = github.context.runNumber.toString()
+  const name: string = await artifactsName()
+  const workflow: string = await workflowName()
+  const metadataUrl: string = new URL(
+    path.join(
+      '/add_metadata/',
+      'github',
+      owner,
+      repo,
+      workflow,
+      runNumber,
+      name
+    ),
+    url
+  )
+    .toString()
+    .concat('/')
+    .replace(/\/$/, '')
+
+  core.debug(metadataUrl)
+  const requestConfig: AxiosRequestConfig = {
+    params: metadata,
+    validateStatus(status: number): boolean {
+      return status === 200
```

Comment:
This is not enough to validate the status of a metadata upload. As the headers are sent before the body, it sends 200 first then logs stuff as it process them and if something fails, it will be mentioned. If everything is good, it will end with the final line "PASSED" in the body. This has to be checked.

Reply:
Okk I'll make the according modification

Reply:
done